src/doppler_env/__init__.py

```diff
@@ -1,26 +1,78 @@
 """
 Imported by doppler_env.pth Automatically inject Doppler secrets as environment variables when spawning a Python process if the `DOPPLER_ENV` environment variable is set
 """
+from base64 import b64encode
+import io
 import os
+import subprocess
+import sys
+import urllib.request
+import urllib.error
 
+from dotenv import load_dotenv
 
-def doppler_secrets_env():
-    import io
-    import subprocess
-    from dotenv import load_dotenv
 
-    DOPPLER_ENV_COMMAND = os.environ.get(
-        'DOPPLER_ENV_COMMAND', 'doppler secrets download --no-file --format env'
-    )
+def log(message):
+    print('[doppler-env]: {}'.format(message))
+
+
+def print_debug_info(doppler_token=None, project=None, config=None):
+    doppler_token and log('Token: {}'.format(doppler_token))
+    project and log('Project: {}'.format(project))
+    config and log('Config: {}'.format(config))
+    log('Python path: {}'.format(sys.executable))
+    log('Working directory: {}'.format(os.getcwd()))
+
+
+def fetch_cli():
+    command = os.environ.get('DOPPLER_ENV_COMMAND', 'doppler secrets download --no-file --format env')
+
+    try:
+        subprocess.check_output('doppler', stderr=subprocess.STDOUT).decode('utf-8')
+    except FileNotFoundError:
+        log('The Doppler CLI is not installed. See https://docs.doppler.com/docs/install-cli')
+        return
 
     # If non-zero exit code, catch Python exception so only output is stderr from Doppler CLI
     try:
-        env_vars = subprocess.check_output(DOPPLER_ENV_COMMAND.split()).decode('utf-8')
-    except subprocess.CalledProcessError as err:
-        exit(err.returncode)
+        return subprocess.check_output(command.split()).decode('utf-8')
+    except subprocess.CalledProcessError:
+        print_debug_info()
 
-    load_dotenv(stream=io.StringIO(env_vars))
+
+def fetch_api(doppler_token, project=None, config=None):
+    if ('dp.ct' in doppler_token or 'dp.pt' in doppler_token) and (project is None or config is None):
+        log(
+            'Error: DOPPLER_PROJECT and DOPPLER_CONFIG environment variables must be set if using a CLI or Personal Token'
+        )
+        print_debug_info()
+        return
+
+    auth_header = b'Basic %s' % b64encode('{}:'.format(doppler_token).encode('ascii'))
+    headers = {'User-Agent': 'python-doppler-env', 'Authorization': auth_header}
+    url = 'https://api.doppler.com/v3/configs/config/secrets/download?format=env'
+    if 'dp.st' not in doppler_token and project and config:
+        url = '{url}&project={project}&config={config}'.format(url=url, project=project, config=config)
```

Comment:
Instead of manually building these query strings, it might be worth it to build with urllib:

```python
from urllib import parse
parse.urlencode({'format': format, 'project': project, 'config': config})
# 'format=json&project=backend&config=dev'
```

One big advantage here is that it'll URL-escape the values

Reply:
Done!